src/test/java/org/sqlite/SerializeTest.java

```diff
@@ -0,0 +1,176 @@
+
+
+public class SerializeTest {
+
+    private ByteBuffer serialize() throws SQLException {
+        try (SQLiteConnection connection = (SQLiteConnection) DriverManager.getConnection("jdbc:sqlite:")) {
+            execute(connection, "ATTACH ? AS ?", ":memory:", "a_schema");
+            execute(connection, "CREATE TABLE a_schema.a_table (x integer)");
+            execute(connection, "INSERT INTO a_schema.a_table (x) values (?)", 1007);
+            return connection.serialize("a_schema");
+        }
+    }
+
+    @Test
+    public void testSerializeDeserialize() throws SQLException {
+        ByteBuffer bb = serialize();
+        deserializeAndAssert(bb);
+    }
+
+    private void deserializeAndAssert(ByteBuffer bb) throws SQLException {
+        try (SQLiteConnection connection = (SQLiteConnection) DriverManager.getConnection("jdbc:sqlite:")) {
+            execute(connection, "ATTACH ? AS ?", ":memory:", "another_schema");
+            connection.deserialize("another_schema", bb);
+            assertThat(fetch(connection, "SELECT * FROM another_schema.a_table")).isEqualTo(1007);
+        }
+    }
+
+    @Test
+    public void testSerializeByteArrayDeserialize() throws SQLException {
+        ByteBuffer bb = serialize();
+        assertThat(bb.isDirect()).isTrue();
+        byte[] buff = new byte[bb.remaining()];
+        bb.get(buff);
+        try {
+            deserializeAndAssert(ByteBuffer.wrap(buff));
+            fail("exception expected");
+        } catch (IllegalArgumentException ignore) {
+            // has to be a direct buffer
+        }
+
+        bb = ByteBuffer.allocateDirect(buff.length);
+        bb.put(buff);
+        deserializeAndAssert(bb);
+    }
+
+    @Test
+    public void testGrow() throws SQLException {
+        ByteBuffer bb = serialize();
+        int size = bb.remaining();
+        try (SQLiteConnection connection = (SQLiteConnection) DriverManager.getConnection("jdbc:sqlite:")) {
+            execute(connection, "ATTACH ? AS ?", ":memory:", "a_schema");
+            connection.deserialize("a_schema", bb);
+            connection.setAutoCommit(false);
+            try (PreparedStatement ps = prepare(connection, "INSERT INTO a_schema.a_table (x) values (?)", 1)) {
+                for (int i=0;i<10000;++i) {
+                    ps.execute();
+                }
+            }
+            connection.commit();
+            bb = connection.serialize("a_schema");
+            int newSize = bb.remaining();
+            assertThat(newSize).isGreaterThan(size);
+        }
+
+        try (SQLiteConnection connection = (SQLiteConnection) DriverManager.getConnection("jdbc:sqlite:")) {
+            execute(connection, "ATTACH ? AS ?", ":memory:", "a_schema");
+            connection.deserialize("a_schema", bb);
+            assertThat(fetch(connection, "SELECT count(*) FROM a_schema.a_table")).isEqualTo(10001);
+        }
+    }
+
+    @Test
+    public void testMultiFirstTimeSerialize() throws SQLException {
```

Comment:
what does that test ? there's no assertions

Reply:
Added assert that size doesn't change across runs